options.go

```diff
@@ -0,0 +1,72 @@
+package kbucket
+
+import (
+	"fmt"
+	"time"
+)
+
+// Option is the Routing Table functional option type.
+type Option func(*Options) error
+
+// Options is a structure containing all the functional options that can be used when constructing a Routing Table.
+type Options struct {
+	TableCleanup struct {
```

Comment:
Somewhat OT: @raulk do we want to continue making the details of options structs public or make them private? IMO, users shouldn't be setting these directly (i.e., they should use pre-defined `Option`s).

Reply:
If we have option funcs for all of these, and we don’t want to provide a single “rule them all” option func where the user supplies the struct, let’s make them private.

Reply:
Made the change.